app/code/community/ShipStream/Sync/Model/Order/Shipment/Api.php

```diff
@@ -124,4 +131,111 @@ public function createWithTracking($orderIncrementId, $data)
 
         return $shipment->getIncrementId();
     }
+
+    /**
+     * Retrieve Shipped Order Item Qty from Shipstream shipment packages
+     * @param $order
+     * @param $data
+     * @return array
+     */
+    protected function _getShippedItemsQty($order, $data)
+    {
+        $orderItems = [];
+        $itemShippedQty = [];
+
+        // Get order item reference Ids from shipment data
+        foreach ($data['items'] as $dataItem) {
+            $orderItems[$dataItem['order_item_id']] = $dataItem['order_item_ref'];
+        }
+
+        // Payload data to create shipment in openmage for only items shipped from shipstream
+        foreach ($data['packages'] as $package) {
+            foreach ($package['items'] as $item) {
+                $key = $orderItems[$item['order_item_id']];
+                if (isset($itemShippedQty[$key])) {
+                    $itemShippedQty[$key] = $itemShippedQty[$key] + floatval($item['order_item_qty']);
+                }
+                else {
+                    $itemShippedQty[$key] = floatval($item['order_item_qty']);
+                }
+            }
+        }
+
+        // Discard items that are partially shipped
+        foreach ($itemShippedQty as $item_id => $ordered_qty) {
+            // Handling fractional BOM items shipped qty
+            $fraction = fmod($ordered_qty, 1);
+            $wholeNumber = intval($ordered_qty);
+            if ($fraction >= 0.88) {
```

Comment:
Why 0.88? I think the floating point error should be very small, probably 0.9999 would be good enough to catch all floating point arithmetic errors but still allow for some edge cases like 29/30 of the total shipped.